src/commands/init.js

```
@@ -83,84 +222,323 @@ Examples:
+    return
+  }
+
+  // Clone the example subgraph repository
+  let cloned = await withSpinner(
+    `Cloning example subgraph`,
+    `Failed to clone example subgraph`,
+    async spinner => {
+      await system.run(
+        `git clone http://github.com/graphprotocol/example-subgraph ${directory}`
+      )
+      return true
+    }
+  )
+  if (!cloned) {
+    process.exitCode = 1
+    return
+  }
+
+  // Update package.json to match the subgraph name
+  let prepared = await withSpinner(
+    `Update subgraph name and commands in package.json`,
+    `Failed to update subgraph name and commands in package.json`,
+    async spinner => {
+      try {
+        // Load package.json
+        let pkgJsonFilename = filesystem.path(directory, 'package.json')
+        let pkgJson = await filesystem.read(pkgJsonFilename, 'json')
+
+        pkgJson.name = getSubgraphBasename(subgraphName)
+        Object.keys(pkgJson.scripts).forEach(name => {
+          pkgJson.scripts[name] = pkgJson.scripts[name].replace('example', subgraphName)
+        })
+        delete pkgJson['license']
+        delete pkgJson['repository']
+
+        // Write package.json
+        await filesystem.write(pkgJsonFilename, pkgJson, { jsonIndent: 2 })
+        return true
+      } catch (e) {
+        print.error(`Failed to preconfigure the subgraph: ${e}`)
+        filesystem.remove(directory)
+        return false
+      }
+    }
+  )
+  if (!prepared) {
+    process.exitCode = 1
+    return
+  }
+
+  // Initialize a fresh Git repository
+  let repo = await initRepository(toolbox, directory)
+  if (repo !== true) {
+    process.exitCode = 1
+    return
+  }
+
+  // Install dependencies
+  let installed = await installDependencies(toolbox, directory, commands.install)
+  if (installed !== true) {
+    process.exitCode = 1
+    return
+  }
+
+  // Run code-generation
+  let codegen = await runCodegen(toolbox, directory, commands.codegen)
+  if (codegen !== true) {
+    process.exitCode = 1
```

Setting the exit code and then returning doesn't seem like the best practice, perhaps bubble up an exception and then set the exit code when catching it?

---

The thing is: `runCodegen` already runs a spinner that will log an error if code generation fails. I didn't want to output two identical errors. We could throw an error here and then not print it at the caller level but that is not better IMHO.